fuzz/fuzz_targets/spectests.rs

```diff
@@ -0,0 +1,33 @@
+#![no_main]
+
+use arbitrary::Arbitrary;
+use libfuzzer_sys::fuzz_target;
+use wasmtime::{Config, Engine, Store};
+use wasmtime_fuzzing::generators::OptLevel;
+use wasmtime_wast::WastContext;
+
+include!(concat!(env!("OUT_DIR"), "/spectests.rs"));
+
+#[derive(Arbitrary, Debug)]
+pub struct SpecCompliantConfig {
+    opt_level: OptLevel,
+    debug_verifier: bool,
+    debug_info: bool,
+    canonicalize_nans: bool,
+    spectest: usize,
+}
+
+fuzz_target!(|config: SpecCompliantConfig| {
+    let mut cfg = Config::new();
+    cfg.debug_info(config.debug_info)
+        .cranelift_nan_canonicalization(config.canonicalize_nans)
+        .cranelift_debug_verifier(config.debug_verifier)
+        .cranelift_opt_level(config.opt_level.to_wasmtime());
+
+    let (file, contents) = FILES[config.spectest % FILES.len()];
```

Comment:
Taking the mod won't result in a uniform distribution. Instead, this should be part of the `Arbitrary` impls:

```rust
struct SpecFile {
    file: &'static str,
    contents: &'static str,
}

impl Arbitrary for SpecFile {
    fn arbitrary(u: &mut Unstructured) -> arbitrary::Result<Self> {
        // NB: this does get a uniform value in the provided range.
        let i = u.int_in_range(0, FILES.len())?;
        let (file, contents) = FILES[i];
        Ok(SpecFile { file, contents })
    }

    fn size_hint(_depth: usize) -> (usize, Option<usize>) {
        (1, Some(std::mem::size_of::<usize>()))
    }
}
```